### PR TITLE
Split the IPAddress and Port

### PR DESCRIPTION
**Pull request by JonghoL**

### CODE REVIEW

**Comment by davidfowl** [comment]
Add some tests

File: src/Microsoft.AspNet.IISPlatformHandler/IISPlatformHandlerMiddleware.cs [hunk 1]
```diff
@@ -91,6 +93,11 @@ private static void UpdateRemoteIp(HttpContext httpContext)
                         httpContext.Request.Headers[XOriginalIPName] = remoteIPString;
                     }
                     httpContext.Connection.RemoteIpAddress = ipFromHeader;
+                    int remotePortFromHeader;
+                    if (int.TryParse(ipString.Substring(portStart+1, ipString.Length - (portStart+1)), out remotePortFromHeader))
```
Comment (davidfowl) [inline src/Microsoft.AspNet.IISPlatformHandler/IISPlatformHandlerMiddleware.cs]: Split this into multiple lines

**Comment by Tratcher** [comment]
Test input:
`127.0.0.1`
`127.0.0.1:80` - Currently broken
`::1` - Regressed by this PR.
`[::1]` - Regressed by this PR.
`[::1]:80` - Currently broken

**Comment by pakrym** [comment]
Fixed in https://github.com/aspnet/IISIntegration/pull/38/files

**Comment by GuardRex** [comment]
@pakrym This might be the wrong spot to ask (I'll move it back to https://github.com/aspnet/IISIntegration/issues/17 if that's best):

1. Forwarded is the RFC now (https://tools.ietf.org/html/rfc7239), that's coming at some point down the road?
2. This all looks to be IPv4, what's the story on IPv6 support? Does the `TryParse` handle it ok?
3. The headers list ... I mean I found a lot more when I looked: X-FORWARDED-FOR, REMOTE_ADDR, HTTP_X_FORWARDED_FOR, HTTP_CLIENT_IP, HTTP_X_FORWARDED, "HTTP_X_CLUSTER_CLIENT_IP, HTTP_FORWARDED_FOR, HTTP_FORWARDED, X_FORWARDED_FOR, CLIENT_IP, X_FORWARDED, X_CLUSTER_CLIENT_IP, FORWARDED_FOR, FORWARDED. I was just wondering if some of these other non-standard headers that we might come up against in the wild should be handed by us ... not by the package?

**Comment by Tratcher** [comment]
@GuardRex For 1 and 3, this middleware only handles headers set by the HttpPlatformHandler. I've asked them about 1 and they said they'd look into it.
For 2, this was PR replaced by https://github.com/aspnet/IISIntegration/pull/38 where IPv6 works fine.

**Comment by GuardRex** [comment]
@Tratcher Gotcha. I saw @pakrym closed https://github.com/aspnet/IISIntegration/issues/17, so that's why I asked.

For #3 though, we should handle funky non-standard headers for IP ourselves. I just want to make sure I know what I can count on with `Connection.RemoteIpAddress` and `Connection.RemotePort`.

**Pull request closed without merging**